### PR TITLE
Implement needed bitfield methods

### PR DESCRIPTION
**Pull request by whyrusleeping**

### CODE REVIEW

**Comment by magik6k** [comment]
Request: `[Un]MarshalJSON` (marshalling as something like `"b64bytes"`)

File: bitfield.go [hunk 1]
```diff
@@ -256,3 +276,236 @@ func (bf *BitField) UnmarshalCBOR(r io.Reader) error {
...
+			}
+		} else {
+			i += r.Len
+		}
+	}
+	return nil
+}
+
+func (bf *BitField) IsSet(x uint64) (bool, error) {
+	iter, err := bf.sum()
```
Comment (magik6k) [inline bitfield.go]: Just checking if `bf.set[x]` is set here instead of doing `sum` would likely be fasterer

File: bitfield.go [hunk 2]
```diff
@@ -256,3 +276,236 @@ func (bf *BitField) UnmarshalCBOR(r io.Reader) error {
...
+		}
+
+		if r.Val {
+			if r.Len <= valsUntilStart {
+				valsUntilStart -= r.Len
+				i += r.Len
+			} else {
+				i += valsUntilStart
+
+				rem := r.Len - valsUntilStart
```
Comment (magik6k) [inline bitfield.go:408]: can't `rem` be 0 / negative?

File: rle/runs.go [hunk 3]
```diff
@@ -117,3 +118,118 @@ func Count(ri RunIterator) (uint64, error) {
...
+
+func (ni *notIter) HasNext() bool {
+	return true
+}
+
+func (ni *notIter) NextRun() (Run, error) {
+	if !ni.it.HasNext() {
+		return Run{
+			Val: true,
+			Len: 10000000000, // close enough to infinity
```
Comment (magik6k) [inline rle/runs.go:220]: That limits usable SectorNumbers, doesn't it?

File: rle/runs.go [hunk 4]
```diff
@@ -204,3 +204,32 @@ func (ri *RunSliceIterator) NextRun() (Run, error) {
 	ri.i++
 	return out, nil
 }
+
+type notIter struct {
+	it RunIterator
+}
+
+func (ni *notIter) HasNext() bool {
+	return true
```
Comment (Kubuxu) [inline rle/runs.go:213]: I think `HasNext` always returning true might lead to trouble, or maybe not.

File: bitfield.go (hunk 2 above)
Comment (whyrusleeping) [reply bitfield.go:408]: no, we guard against that in the outer if statement, `r.Len > valsUntilStart`

File: rle/runs.go (hunk 4 above)
Comment (whyrusleeping) [reply rle/runs.go:213]: Well, in the usecase I have for it, its fine. Ideally we can mark runs as being of infinite length, and things can handle that specially

File: rle/runs.go (hunk 3 above)
Comment (whyrusleeping) [reply rle/runs.go:220]: Nah, It just keeps returning runs of a billion-ish forever. so if you run out of this one, it just returns another

**Pull request merged**